ops/model.py

```diff
@@ -810,6 +895,281 @@ def __init__(self, name: str, address_info: '_AddressDict'):
+        :meth:`Model.get_secret` with an ID but no label.
+        """
+        return self._label
+
+    def get_content(self, *, refresh: bool = False) -> Dict[str, str]:
+        """Get the secret's content.
+
+        Args:
+            refresh: If true, fetch the latest revision's content and tell
+                Juju to update to tracking that revision. The default is to
+                get the content of the currently-tracked revision.
+        """
+        if refresh or self._content is None:
+            self._content = self._backend.secret_get(
+                id=self.id, label=self.label, refresh=refresh)
+        return self._content
+
+    def peek_content(self) -> Dict[str, str]:
+        """Get the content of the latest revision of this secret.
+
+        This returns the content of the latest revision without updating the
+        tracking.
+        """
+        return self._backend.secret_get(id=self.id, label=self.label, peek=True)
+
+    def get_info(self) -> SecretInfo:
+        """Get this secret's information (metadata).
+
+        Only secret owners can fetch this information.
+        """
+        return self._backend.secret_info_get(id=self.id, label=self.label)
+
+    def set_content(self, content: Dict[str, str]):
+        """Update the content of this secret.
+
+        This will create a new secret revision, and notify all units tracking
+        the secret (the "consumers") that a new revision is available with a
+        :class:`ops.charm.SecretChangedEvent`.
+
+        Args:
+            content: A key-value mapping containing the payload of the secret,
+                for example :code:`{"password": "foo123"}`.
+        """
+        self._validate_content(content)
+        if self._id is None:
+            self._id = self.get_info().id
+        self._backend.secret_set(typing.cast(str, self.id), content=content)
+
+    def set_info(self, *,
+                 label: Optional[str] = None,
+                 description: Optional[str] = None,
+                 expire: Optional[Union[datetime.datetime, datetime.timedelta]] = None,
+                 rotate: Optional[SecretRotate] = None):
+        """Update this secret's information (metadata).
+
+        This will create a new secret revision, and notify all units tracking
+        the secret (the "consumers") that a new revision is available with a
+        :class:`ops.charm.SecretChangedEvent`.
+
+        Args:
+            label: New label to apply.
+            description: New description to apply.
+            expire: New expiration time (or timedelta from now) to apply.
+            rotate: New rotation policy to apply. The new policy will take
+                effect only after the currently-scheduled rotation.
+        """
+        if label is None and description is None and expire is None and rotate is None:
+            raise TypeError('Must provide a label, description, expiration time, '
+                            'or rotation policy')
```

Comment:
Actually, setting `label` or `description` to the empty string is the same as not setting them (that's a Juju thing -- the hook tool doesn't distinguish). I spoke with Ian, and we think it would be very rare to want to do this, so let's just postpone that decision for now. (Though it is possible to reset the `rotate` attribute to `SecretRotate.NEVER`.) I've clarified in the docstring that you can't do this.